pkg/controller/core/workload_controller.go

```diff
@@ -241,6 +249,35 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	return ctrl.Result{}, nil
 }
 
+func downSizeJobIfNecessary(wl *kueue.Workload) bool {
+	statusUpdate := false
+	podSetSize := len(wl.Spec.PodSets)
+	for i := 1; i < podSetSize; i++ {
+		if ptr.Deref(wl.Status.Admission.PodSetAssignments[i].Count, 0) > wl.Spec.PodSets[i].Count {
+			// Get Resource Requests and Usage values
+			originalResourceRequests := wl.Spec.PodSets[i].Template.Spec.Containers[0].Resources.Requests
```

Comment:
What about requests from other containers?

I think you can use this function https://github.com/kubernetes-sigs/kueue/blob/3347afb5a8681f4fd2a5f3b1a5be8c5c0ebac488/pkg/util/limitrange/limitrange.go#L85